node/src/components/fetcher/metrics.rs

```diff
@@ -0,0 +1,57 @@
+use prometheus::{IntCounter, Registry};
+
+#[derive(Debug)]
+pub struct FetcherMetrics {
+    /// Number of fetch requests that found an item in the storage.
+    pub(super) found_in_storage: IntCounter,
+    /// Number of fetch requests that fetcher an item from peer.
+    pub(super) found_on_peer: IntCounter,
+    /// Number of fetch requests that timed out.
+    pub(super) timeouts: IntCounter,
+    /// Reference to the registry for unregistering.
+    registry: Registry,
+}
+
+impl FetcherMetrics {
+    pub fn new(name: &str, registry: &Registry) -> Result<Self, prometheus::Error> {
+        let found_in_storage = IntCounter::new(
+            format!("{}_found_in_storage", name),
+            format!(
+                "number of fetch requests that found {} in the storage.",
+                name
+            ),
+        )?;
+        let found_on_peer = IntCounter::new(
+            format!("{}_found_on_peer", name),
+            format!("number of fetch requests that fetched {} from peer.", name),
+        )?;
+        let timeouts = IntCounter::new(
+            format!("{}_timeouts", name),
+            format!("number of {} fetch requests that timed out", name),
+        )?;
+        registry.register(Box::new(found_in_storage.clone()))?;
+        registry.register(Box::new(found_on_peer.clone()))?;
+        registry.register(Box::new(timeouts.clone()))?;
+
+        Ok(FetcherMetrics {
+            found_in_storage,
+            found_on_peer,
+            timeouts,
+            registry: registry.clone(),
+        })
+    }
+}
+
+impl Drop for FetcherMetrics {
+    fn drop(&mut self) {
+        self.registry
+            .unregister(Box::new(self.found_in_storage.clone()))
+            .expect("did not expect deregistering found_in_storage to fail");
```

Comment:
I know a lot of the other components are doing it this way, but could we use `unwrap_or_else` with a warning printed rather than a panic?  Similar to how e.g. linear chain handles this.

Reply:
Done in [`9960350` (#925)](https://github.com/CasperLabs/casper-node/pull/925/commits/99603501fb81193ac6820dc9eac965ab73cb98b8).